app/apollo/test/subscriptions.fga.spec.js

```diff
@@ -0,0 +1,659 @@
+  };
+  await models.Cluster.create(testCluster1);
+};
+
+const assignClusterGroups = async ( token, orgId, groupUUIDs, clusterUUID ) => {
+  const methodName = 'assignClusterGroups (subscriptions test)';
+  console.log( `${methodName} entry` );
+  const {
+    data: {
+      data: { assignClusterGroups },
+    },
+  } = await groupApi.assignClusterGroups(token, {
+    orgId: orgId,
+    groupUuids: groupUUIDs,
+    clusterIds: [clusterUUID]
+  });
+  expect(assignClusterGroups.modified).to.equal( groupUUIDs.length );
+  await( sleep(1000) ); // Wait to give async RBAC Sync time to complete
+  console.log( `${methodName} exit` );
+};
+
+const sleep = async ( ms ) => {
+  return new Promise(resolve => {
+    setTimeout(resolve, ms);
+  });
+};
+
+describe('subscription graphql test suite', () => {
+  before(async () => {
+    process.env.NODE_ENV = 'test';
+    rbacSync.testMode(true); // Must be set to trigger/test RBAC Sync
+
+    mongoServer = new MongoMemoryServer( { binary: { version: '4.2.17' } } );
+    await mongoServer.start();
+    const mongoUrl = mongoServer.getUri();
+    console.log(`subscriptions.spec.js in memory test mongodb url is ${mongoUrl}`);
+
+    myApollo = await apollo({
+      mongo_url: mongoUrl,
+      graphql_port: graphqlPort,
+    });
+
+    await createOrganizations();
+    await createUsers();
+    await createGroups();
+    await createChannels();
+    await createVersions();
+    await createSubscriptions();
+    await createClusters();
+
+    fgaToken01 = await signInUser(models, authApi, fgaUser01Data);
+    fgaToken02 = await signInUser(models, authApi, fgaUser02Data);
+  }); // before
+
+  after(async () => {
+    await myApollo.stop(myApollo);
+    GraphqlPubSub.deleteInstance();
+    await mongoServer.stop();
+  }); // after
+
+  // subscriptions for fgaUser01
+  it('fgaUser01 has authorization to get ALLOWED subscription 1', async () => {
+    let response;
+    try {
+      response = await subscriptionApi.subscriptions(fgaToken01, {
+        orgId: org01._id,
+      });
+      const subscriptions = response.data.data.subscriptions;
+      expect(subscriptions).to.have.length(1);
+      expect(subscriptions[0].name).to.equal(testSubscription1.name);
+      for( const subscription of subscriptions ) {
+        expect( subscription.identitySyncStatus, 'subscription did not include identitySyncStatus' ).to.exist;
+        expect( subscription.identitySyncStatus.syncedCount, 'subscription identitySyncStatus.syncedCount should be zero' ).to.equal(0);
+        expect( subscription.identitySyncStatus.failedCount, 'subscription identitySyncStatus.failedCount should be zero' ).to.equal(0);
+        expect( subscription.identitySyncStatus.pendingCount, 'subscription identitySyncStatus.pendingCount should be zero' ).to.equal(0);
+        expect( subscription.identitySyncStatus.unknownCount, 'subscription identitySyncStatus.unknownCount should be zero' ).to.equal(0);
+      }
```

Comment:
```suggestion
        expect( subscriptions[0].identitySyncStatus, 'subscription did not include identitySyncStatus' ).to.exist;
        expect( subscriptions[0].identitySyncStatus.syncedCount, 'subscription identitySyncStatus.syncedCount should be zero' ).to.equal(0);
        expect( subscriptions[0].identitySyncStatus.failedCount, 'subscription identitySyncStatus.failedCount should be zero' ).to.equal(0);
        expect( subscriptions[0].identitySyncStatus.pendingCount, 'subscription identitySyncStatus.pendingCount should be zero' ).to.equal(0);
        expect( subscriptions[0].identitySyncStatus.unknownCount, 'subscription identitySyncStatus.unknownCount should be zero' ).to.equal(0);
```
This _works_, but implies that it's really expecting a multi-element array of subscriptions.  It's already stated `expect(subscriptions).to.have.length(1)` however -- this code **knows** it's just one, so it should just do `subscriptions[0]` without the for loop.

Reply:
Same comment applies elsewhere -- anywhere it's getting just a single result and checking "length is expected to be 1" first, just use `[0]` afterwards.

Reply:
Resolved